openmicroscopy/openmicroscopy
# CLI hyphen options

This PR should extend the effort towards the usage of hyphens in favor of underscores in the command-line options.

The CLI import (Python and Java) commands are modified  to:
- the `--annotation-ns`,  `--annotation-text`, `--annotation-link`, `--no-thumbnails` and `--checksum-algorithm` options are implemented (Java and Python side for the first 3 and Java-side for the last two) and the corresponding underscore options are deprecated
- the `CommandLineInterface` logic is refactor to simplify handling of conflicting arguments
- `cli_import.py` and `clitest/test_import.py` are merged into a single class for clarity
- integration tests are extended to cover import with annotations using Java/Python, deprecated/non-deprecated arguments as well as multiple annotations (Java-only)
- integration tests are extended to cover import with different checksum algorithms
- integration tests are extended to cover import with disabled thumbnailing
- the links to the CLI import documentation are modified to point at the new URL (will be published with 5.1.0-m5 release)

To test this PR:
- review the output of `bin/omero import -h`
- review the integration test and check they are passing
- check passing conflicting Java arguments aborts and returns the CLI usage, e.g.

  ````
bin/omero import test.fake -- --no-thumbnails --no_thumbnails
 ````

--no-rebase
/cc @ximenesuk
